tests/integration/infrastructure/scalingo-api_test.js

```diff
@@ -162,5 +162,164 @@ describe('scalingo-api', () => {
+        it('should return only the first N ones, sorted by descending duration', async () => {
+          // given
+          const addonId = 'addonId';
+          const token = 'token';
+          const getCredentials = () => {
+            return { addonId, token };
+          };
+          const baseURL = 'https://db-api.REGION.scalingo.com';
+
+          const queries = [
+            {
+              query:
+                'SELECT usename, query, state, query_start, pid FROM pg_stat_activity',
+              state: 'active',
+              username: 'admin',
+              query_start: '2022-09-28T14:20:05.328404Z',
+              pid: 19296,
+              query_duration: 1,
+            },
+            {
+              query:
+                'SELECT * FROM users',
+              state: 'active',
+              username: 'admin',
+              query_start: '2022-09-28T11:20:05.328404Z',
+              pid: 19297,
+              query_duration: 2,
+            },
+            {
+              query:
+                'SELECT * FROM users',
+              state: 'active',
+              username: 'admin',
+              query_start: '2022-09-28T11:20:05.328404Z',
+              pid: 19291,
+              query_duration: 4,
+            },
+            {
+              query:
+                'SELECT * FROM users',
+              state: 'active',
+              username: 'admin',
+              query_start: '2022-09-28T11:20:05.328404Z',
+              pid: 19298,
+              query_duration: 3,
+            },
+          ];
+          const scalingoResponse = {
+            result: queries,
+          };
+          const scalingoApp = 'application';
+          nock(baseURL)
+            .post(`/api/databases/${addonId}/action`, {
+              action_name: 'pg-list-queries',
+            })
+            .reply(200, scalingoResponse);
+
+          // when
+          const response = await getRunningQueries(scalingoApp, getCredentials);
+
+          // then
+          expect(response.topNQueries).to.deep.equal( [
+            {
+              query:
+                'SELECT * FROM users',
+              state: 'active',
+              username: 'admin',
+              query_start: '2022-09-28T11:20:05.328404Z',
+              pid: 19291,
+              query_duration: 4,
+            },
+            {
+              query:
+                'SELECT * FROM users',
+              state: 'active',
+              username: 'admin',
+              query_start: '2022-09-28T11:20:05.328404Z',
+              pid: 19298,
+              query_duration: 3,
+            },
+            {
+            query:
+              'SELECT * FROM users',
+            state: 'active',
+            username: 'admin',
+            query_start: '2022-09-28T11:20:05.328404Z',
+            pid: 19297,
+            query_duration: 2,
+          }]);
+        });
+      });
+      describe('when there is more less queries than topN', () => {
```

Comment:
when there is less  than N active queries

Reply:
Corrigé